py/surveyqa/summary.py

```diff
@@ -715,11 +713,9 @@ def total_exptime_dgb(program, color):
             program: String of the desired program name
             color: Color of histogram
         '''
-        a = exposures_nocalib.group_by("TILEID").groups.aggregate(sum_or_first)
-        w = a["PROGRAM"] == program
-        if not any(w):
-            return
-        a = a[w]
+        thisprogram = (exposures_nocalib["PROGRAM"] == program)
+        a = exposures_nocalib["TILEID", "EXPTIME"][thisprogram].group_by("TILEID").groups.aggregate(np.sum)
```

Comment:
Two optimizations here:
  1. filter by PROGRAM before doing the group_by and aggregate, thus grouping/aggregating on fewer items.  This also enables (2):
  2. don't include the "PROGRAM" column in the group so that we can aggregate on the faster `np.sum` function instead of a custom `sum_or_first` that required a type check and an if statement before getting to `np.sum`.